### PR TITLE
Rerun a flaky test

### LINKED ISSUE #638: Fix flaky docker celery tests
Celery docker tests randomly fail with the following output:

```
=================================== FAILURES ===================================
_________________________ test_custom_tracer_provider __________________________

celery_app = <Celery celery.tests at 0x7f88b2d3cbe0>
memory_exporter = <opentelemetry.sdk.trace.export.in_memory_span_exporter.InMemorySpanExporter object at 0x7f88b8305790>

    def test_custom_tracer_provider(celery_app, memory_exporter):
        @celery_app.task
        def fn_task():
            return 42
    
        resource = resources.Resource.create({})
        tracer_provider = TracerProvider(resource=resource)
        span_processor = export.SimpleSpanProcessor(memory_exporter)
        tracer_provider.add_span_processor(span_processor)
    
        trace_api.set_tracer_provider(tracer_provider)
    
        CeleryInstrumentor().uninstrument()
        CeleryInstrumentor().instrument(tracer_provider=tracer_provider)
    
        fn_task.delay()
    
        spans_list = memory_exporter.get_finished_spans()
>       assert len(spans_list) == 1
E       assert 2 == 1
E         +2
E         -1

celery/test_celery_functional.py:579: AssertionError
```

We need to investigate this and fix th test so it always passes.

### PR DESCRIPTION
**Pull request by ocelotl**
Workaround for #638

### CODE REVIEW

**Comment by owais** [comment]
This can help but I don't think it should be considered a fix for #638. We should actually fix the flaky test to close that ticket.

**Review by owais — APPROVED** [review]

File: tests/opentelemetry-docker-tests/tests/celery/test_celery_functional.py [hunk 1]
```diff
@@ -558,6 +558,8 @@ class CelerySubClass(CelerySuperClass):
     ) == async_run_span.attributes.get(SpanAttributes.MESSAGING_MESSAGE_ID)
 
 
+# FIXME find a permanent solution for the flakiness of this test
```
Comment (codeboten) [inline tests/opentelemetry-docker-tests/tests/celery/test_celery_functional.py:561]: If the fix for the issue includes a FIXME, then I agree w/ @owais that this doesn't fix #638 😄

**Review by codeboten — APPROVED** [review]
I think its fine to skip the flaky test, but the PR should not be marked as a fix for the problem.

**Review by lzchen — APPROVED** [review]

File: tests/opentelemetry-docker-tests/tests/celery/test_celery_functional.py (hunk 1 above)
Comment (ocelotl) [reply tests/opentelemetry-docker-tests/tests/celery/test_celery_functional.py:561]: I have updated the description of this PR :+1:

**Pull request merged**